book/build.py

```diff
@@ -0,0 +1,58 @@
+import subprocess
+from pathlib import Path
+import urllib.parse
+
+
+SCRIPT_DIR = Path(__file__).parent.resolve()
+
+
+def generate_example(filename, title, strip_asserts=True, insert_run_link=True):
+    path_in = SCRIPT_DIR.parent / "examples" / f"{filename}.nbt"
+    path_out = SCRIPT_DIR / "src" / f"example-{filename}.md"
+    print(path_in)
+    print(path_out)
+
+    code = []
+    with open(path_in, "r") as fin:
+        for line in fin:
+            if not (strip_asserts and "assert_eq" in line):
+                code.append(line)
+
+    url = f"https://numbat.dev/?q={urllib.parse.quote_plus(''.join(code))}"
+
+    with open(path_out, "w") as fout:
+        fout.write("<!-- This file is autogenerated! Do not modify it -->\n")
+        fout.write("\n")
+        fout.write(f"# {title}\n")
+        if insert_run_link:
+            fout.write(f'<a href="{url}"><i class="fa fa-play"></i> Run this example</a>\n')
+        fout.write("\n")
+        fout.write("``` numbat\n")
+        fout.writelines(code)
+        fout.write("```\n")
+
+
+generate_example("acidity", "Acidity")
+generate_example("barometric_formula", "Barometric formula")
+generate_example("body_mass_index", "Body mass index")
+generate_example("factorial", "Factorial", strip_asserts=False)
+generate_example("medication_dosage", "Medication dosage")
+generate_example("molarity", "Molarity")
+generate_example("musical_note_frequency", "Musical note frequency")
+generate_example("paper_size", "Paper sizes")
+generate_example("pipe_flow_rate", "Flow rate in a pipe")
+generate_example("population_growth", "Population growth")
+generate_example("recipe", "Recipe")
+generate_example("voyager", "Voyager")
+generate_example("xkcd_687", "XKCD 687")
+generate_example("xkcd_2585", "XKCD 2585")
+generate_example("xkcd_2812", "XKCD 2812")
+generate_example("workhours", "Workhours")
```

Comment:
The "workhours" example was missing in 'build.sh', I would think this was not intentional, right?

Reply:
It was. Notice that there are a lot more examples in `examples/` that are not part of the documentation. I only hand-selected a few of them. I removed this line again.

Reply:
Yes, I noticed that, but it is the only example that is not present in `build.sh`/`build.py` but has an [associated file](https://github.com/sharkdp/numbat/blob/4709bc3d9b6a546a0d90b77578c54bd41faf5167/book/src/example-workhours.md) in `/book/src/`. Maybe that file should be removed too in this case?

Reply:
Oh, right.  I guess it was part of the documentation at some point but then I removed it. I'll remove that file. Thank you.